### PR TITLE
Jetpack connect: Refactor JetpackPlansLanding

### PR DESCRIPTION
**Pull request by sirreal**
Pass linter (es6 class)

Lint fixes required for: Separate jetpack-connect from signup #12991

To test:

* Use this branch.
* Test the following paths:
  * http://calypso.localhost:3000/jetpack/connect/store
  * http://calypso.localhost:3000/jetpack/connect/store/monthly
  * http://calypso.localhost:3000/jetpack/connect/vaultpress
  * http://calypso.localhost:3000/jetpack/connect/vaultpress/monthly
  * http://calypso.localhost:3000/jetpack/connect/akismet
  * http://calypso.localhost:3000/jetpack/connect/akismet/monthly
* Test buttons, render etc.
* Ensure the `/monthly` versions display monthly pricing
* Ensure there are no regressions. Watch the console for any errors/warnings.

### CODE REVIEW

File: client/signup/jetpack-connect/plans-landing.jsx [hunk 1]
```diff
@@ -58,7 +56,7 @@ const PlansLanding = React.createClass( {
 			</div>
 		);
 	}
-} );
+}
 
 export default connect(
 	() => {
```
Comment (oskosk) [inline client/signup/jetpack-connect/plans-landing.jsx]: The mapDispatchToProps declaration can probably simplified:

From a function:
```js harmony
	( dispatch ) => {
		return Object.assign( {},
			bindActionCreators( { selectPlanInAdvance }, dispatch ),
			{
				recordTracksEvent( eventName, props ) {
					dispatch( recordTracksEvent( eventName, props ) );
				}
			}
		);
	}
```

to a plain object
```
{
  selectPlanInAdvance,
  recordTracksEvent
}

```

File: client/signup/jetpack-connect/plans-landing.jsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 /**
  * External dependencies
  */
+import React, { Component } from 'react';
```
Comment (oskosk) [inline client/signup/jetpack-connect/plans-landing.jsx]: You can bring in `PropTypes` as a named import too.

File: client/signup/jetpack-connect/plans-landing.jsx (hunk 1 above)
Comment (sirreal) [inline client/signup/jetpack-connect/plans-landing.jsx]: I'd prefer to get this in a PR. I'm trying to limit the scope, I've had several PRs really get away from me in terms of size. There are a few fixes that can be refactored here, but if you don't think it's a blocker, I'd rather leave it out of this.

Comment (oskosk) [inline client/signup/jetpack-connect/plans-landing.jsx]: wfm!

File: client/signup/jetpack-connect/plans-landing.jsx [hunk 3]
```diff
@@ -16,28 +16,26 @@ import QueryPlans from 'components/data/query-plans';
 
 const CALYPSO_JETPACK_CONNECT = '/jetpack/connect';
 
-const PlansLanding = React.createClass( {
+class PlansLanding extends Component {
 
-	propTypes: {
+	static propTypes = {
 		sites: React.PropTypes.object,
```
Comment (oskosk) [inline client/signup/jetpack-connect/plans-landing.jsx]: is the `sites` prop still used here or in the child component ?

Comment (sirreal) [inline client/signup/jetpack-connect/plans-landing.jsx]: This is a big part of the other fixes I mentioned. There a very weird `connect` `mapStateToProps` going on, that adds props that don't depend on `state` and may or may not be used. But yeah, I'd like to tackle all of that separately.

File: client/signup/jetpack-connect/plans-landing.jsx (hunk 2 above)
Comment (sirreal) [inline client/signup/jetpack-connect/plans-landing.jsx]: 👍  rolled this in @ 10cb6dc29

File: client/signup/jetpack-connect/plans-landing.jsx [hunk 4]
```diff
@@ -16,28 +16,28 @@ import QueryPlans from 'components/data/query-plans';
 
 const CALYPSO_JETPACK_CONNECT = '/jetpack/connect';
 
-const PlansLanding = React.createClass( {
+class PlansLanding extends Component {
```
Comment (tyxla) [inline client/signup/jetpack-connect/plans-landing.jsx]: I'm starting to think you're testing my eyesight :rofl: Or is that part of a grand plan to get this into Calypso's standards? 🤔  😆

File: client/signup/jetpack-connect/plans-landing.jsx [hunk 5]
```diff
@@ -16,28 +16,28 @@ import QueryPlans from 'components/data/query-plans';
...
 
-	propTypes: {
-		sites: React.PropTypes.object,
-		sitePlans: React.PropTypes.object.isRequired,
-		intervalType: React.PropTypes.string
-	},
+	static propTypes = {
+		basePlansPath: PropTypes.string,
```
Comment (tyxla) [inline client/signup/jetpack-connect/plans-landing.jsx]: When adding non-required props, maybe you'll also want to add some `defaultProps` for them?

File: client/signup/jetpack-connect/plans-landing.jsx (hunk 4 above)
Comment (sirreal) [inline client/signup/jetpack-connect/plans-landing.jsx]: 😭  ceafc0b5e

File: client/signup/jetpack-connect/plans-landing.jsx (hunk 5 above)
Comment (sirreal) [inline client/signup/jetpack-connect/plans-landing.jsx]: To be honest, I'm torn about whether this is a good idea. It's much more invasive than working with `propTypes` as it actually affects user-facing behavior.

In this particular case, it seems to make sense to default to `''`, but looking 2 components deeper, `PlansFeaturesMain.defaultProps.basePlansPath = null`! So do I "lift" the default from farther down the tree?

I'd be interested in having a conversation about this.

File: client/signup/jetpack-connect/plans-landing.jsx (hunk 1 above)
Comment (sirreal) [inline client/signup/jetpack-connect/plans-landing.jsx]: Had a change of heart and tackled this here: 0dbb90810

**Comment by sirreal** [comment]
@tyxla @oskosk I did a lot of clean up here after all, that's what you get for giving me thoughtful reviews on my tiny PRs :trollface:

Thanks for looking, this will need another pass now.

**Comment by oskosk** [comment]
LGTM!

**Pull request merged**